### PR TITLE
Implement Part 8 Step 2

### LINKED ISSUE #337: Write Part - 8 Step - 2 & unit test
Implement Step02Controller and write all unit tests. Verify the display name of the test conforms to the pattern of "Part X Step Y". Follow the standardize naming, comments, etc. to be consistent.

### PR DESCRIPTION
**Pull request by gumbelmj**
Resolves #337 

I had to go back and update Part 4 Step 2.  If the user says "no" we aren't supposed to quit, but rather just fail. (I think that might be adjusted in the future though)

### CODE REVIEW

File: src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java [hunk 1]
```diff
@@ -87,23 +87,30 @@ protected void run() throws Throwable {
             getListener().onResult(NL + "Attempt " + attempts);
```
Comment (mmschaefer) [inline src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java:87]: The pattern is to not end with periods on the Step identifications.  Can we update to remove that for consistancy?

File: src/org/etools/j1939_84/controllers/part08/Part08Step02Controller.java [hunk 2]
```diff
@@ -53,18 +67,103 @@
...
+                    QuestionListener questionListener = answer -> {
+                        if (answer == CANCEL || answer == NO) {
+                            addFailure("6.8.2.1.b.ii - User says 'no' and no ECU reported an active DTC");
+                            foundDTC.set(true);
+                        }
+                    };
+
+                    String msg = "No module has reported an active DTC." + NL + "Do you wish to continue?";
+                    getListener().onUrgentMessage(msg,
+                                                  "No Active DTCs Found",
```
Comment (mmschaefer) [inline src/org/etools/j1939_84/controllers/part08/Part08Step02Controller.java:104]: nit: For consistancy, can we lower case "Found" and remove the periods?  The project pattern has been only to use the "?" when an action item is expected from the user.   All other info is not expected to be in full sentance format.

**Review by mmschaefer — CHANGES_REQUESTED** [review]

File: src/org/etools/j1939_84/controllers/part08/Part08Step02Controller.java (hunk 2 above)
Comment (gumbelmj) [reply src/org/etools/j1939_84/controllers/part08/Part08Step02Controller.java:104]: I think the title of the message box should be title case.  [Additionally Microsoft, et. al. agree](https://www.microsoftpressstore.com/articles/article.aspx?p=2228449&seqNum=15#:~:text=For%20menu%20names%2C%20command%20names,is%20inconsistent%2C%20use%20title%20capitalization.&text=However%2C%20do%20not%20capitalize%20the,%2C%20scroll%20bar%2C%20and%20icon.)

File: src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java (hunk 1 above)
Comment (gumbelmj) [reply src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java:87]: I didn't see that in here.  I'll update it.  We need to back and update them in the `PartController`

File: src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java [hunk 3]
```diff
@@ -70,40 +71,46 @@
...
-                                              "No Confirmed and Active DTCs Found",
-                                              QUESTION);
+                    QuestionListener questionListener = answer -> {
+                        if (answer == CANCEL || answer == NO) {
+                            addFailure("6.4.2.1.a.ii - User said 'no' and no ECU reported a confirmed and active DTC");
+                            foundDTC.set(true);
+                        }
+                    };
+
+                    String msg = "No module has reported a confirmed and active DTC." + NL + "Do you wish to continue?";
```
Comment (mmschaefer) [inline src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java:109]: Can we please remove the period?

**Review by mmschaefer — APPROVED** [review]

File: src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java (hunk 3 above)
Comment (gumbelmj) [reply src/org/etools/j1939_84/controllers/part04/Part04Step02Controller.java:109]: No.  Dialogs shown to the user should be in proper English: #653

**Pull request merged**